src/instruments/options/finite_difference_pricer.rs

```diff
@@ -0,0 +1,491 @@
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// RustQuant: A Rust library for quantitative finance tools.
+// Copyright (C) 2023 https://github.com/avhz
+// Dual licensed under Apache 2.0 and MIT.
+// See:
+//      - LICENSE-APACHE.md
+//      - LICENSE-MIT.md
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+use crate::instruments::options::option::{ExerciseFlag, TypeFlag};
+use std::f64::consts::E;
+
+/// Finite difference object
+pub struct FiniteDifferencePricer {
+    /// Spot Price
+    pub initial_price: f64,
+    /// Strike price
+    pub strike_price: f64, 
+    /// Risk free rate
+    pub risk_free_rate: f64, 
+    /// Volatility
+    pub volatility: f64, 
+    /// Days until maturity
+    pub time_to_maturity: u16, 
+    /// Option Type
+    pub type_flag: TypeFlag,
+    /// Option Style
+    pub exercise_flag: ExerciseFlag,
+}
+
+impl FiniteDifferencePricer {
+
+    /// Constructor for FiniteDifferencePricer
+    pub fn new(
+        initial_price: f64,
+        strike_price: f64, 
+        risk_free_rate: f64, 
+        volatility: f64, 
+        time_to_maturity: u16, 
+        type_flag: TypeFlag,
+        exercise_flag: ExerciseFlag
+    ) -> Self {
+
+        let mut variables_with_error: Vec<&str> = vec![];
```

Comment:
All of these checks can just simply be `assert` macros. Ideally it would return a `Result` though.

Reply:
The idea was to collect the "incorrect" parameters into a vector and then display a message to clarify which parameter(s) is(are) causing the issue (As opposed to getting an ambiguous error).

Though, I think using `Result` as you suggested is a better idea.

Reply:
I have reconsidered - using `Result` might just overcomplicate things as we would need to "unpack" the object everytime we create an instance. I have resorted to just using `assert` macros.